adsingestp/utils.py

```diff
@@ -105,3 +148,336 @@ def convert(self):
+                nameparser.config.CONSTANTS.titles.add(s)
+            for s in suffix_names:
+                nameparser.config.CONSTANTS.suffix_acronyms.add(s)
+                nameparser.config.CONSTANTS.suffix_not_acronyms.add(s)
+        except Exception as e:
+            logger.exception("Unexpected error setting up the nameparser")
+            raise BaseException(e)
+
+        # Setup the nameparser package and remove *all* of the preset titles:
+        nameparser.config.CONSTANTS.titles.remove(*nameparser.config.CONSTANTS.titles)
+        nameparser.config.CONSTANTS.suffix_acronyms.remove(
+            *nameparser.config.CONSTANTS.suffix_acronyms
+        )
+        nameparser.config.CONSTANTS.suffix_not_acronyms.remove(
+            *nameparser.config.CONSTANTS.suffix_not_acronyms
+        )
+        nameparser.config.CONSTANTS.string_format = (
+            '{last}, {first} "{nickname}", {suffix}, {title}'
+        )
+
+        # Compile regular expressions
+        self.regex_initial = re.compile(r"\. *(?!,)")
+        self.regex_etal = re.compile(r",? et ?al\.?")
+        self.regex_and = re.compile(r" and ")
+        self.regex_dash = re.compile(r"^-")
+        self.regex_quote = re.compile(r"^'")
+        self.regex_the = re.compile(r"^[Tt]he ")
+        self.regex_author = re.compile(
+            r"^(?P<last_name>[^,]+),\s*(?P<initial0>\S)\w*"
+            + "".join(
+                [
+                    r"(?:\s*(?P<initial{}>\S)\S*)?".format(i + 1)
+                    for i in range(self.max_first_name_initials - 1)
+                ]
+            )
+        )
+
+    def _read_datfile(self, filename):
+        output_list = []
+
+        try:
+            fp = open(filename, "r")
+        except Exception as err:
+            # TODO fix logging
+            logger.exception("Error reading file: %s. Error: %s", filename, err)
+        else:
+            with fp:
+                for line in fp.readlines():
+                    if line.strip() != "" and line[0] != "#":
+                        output_list.append(line.strip())
+        return output_list
+
+    def _extract_collaboration(
+        self, collaboration_str, default_to_last_name, collaborations_params
+    ):
+        """
+        Verifies if the author name string contains a collaboration string
+        The collaboration extraction can be controlled by the dictionary
+        'collaborations_params'.
+        """
+        corrected_collaboration_list = []
+        is_collaboration_str = False
+        try:
+            for keyword in collaborations_params["keywords"]:
+                if keyword in collaboration_str.lower():
+                    is_collaboration_str = True
+                    if collaborations_params["first_author_delimiter"]:
+                        # Based on an arXiv author case: "<tag>Collaboration:
+                        # Name, Author</tag>"
+                        authors_list = collaboration_str.split(
```

Comment:
Not our parser's fault! This is actually in the XML export: https://export.arxiv.org/api/query?id_list=2203.16556

> Re-visit how this works since we see strange results in ADS:
> 
> https://ui.adsabs.harvard.edu/abs/2022arXiv220316556B/abstract